juju/home.go

```diff
@@ -23,5 +28,22 @@ func InitJujuXDGDataHome() error {
 	if err := ssh.LoadClientKeys(osenv.JujuXDGDataHomePath("ssh")); err != nil {
 		return errors.Annotate(err, "cannot load ssh client keys")
 	}
+	expiry := time.Now().UTC().AddDate(10, 0, 0)
+	uuid, _ := utils.NewUUID()
+	certdir := osenv.JujuXDGDataHomePath("x509")
+	if certdir != "" {
+		err := os.MkdirAll(certdir, 0700)
+		if err == nil {
+			pem, key, err := cert.NewCA("client", uuid.String(), expiry, 2048)
```

Comment:
as axw pointed, this plans to be a default certificate to be able to tackle the LXD api on a remote endpoint. This can easily be used in any other provider that requires x509 certificates to encrypt communications. The idea in the code is to generate it once, when generating a new config, if it does make the delay in further executions this might be a problem to check now. Let me know if that is the case. Thanks!